api/src/main/java/com/qaengine/controllers/AnswerController.java

```diff
@@ -0,0 +1,92 @@
+package com.qaengine.controllers;
+
+import com.qaengine.database.AnswerRepository;
+import com.qaengine.database.QuestionRepository;
+import com.qaengine.exceptions.ResourceNotFoundException;
+import com.qaengine.lib.HelperFunctions;
+import com.qaengine.models.Answer;
+import com.qaengine.models.Question;
+import com.qaengine.models.inputs.AnswerInput;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@RestController
+public class AnswerController {
+    @Autowired
+    QuestionRepository questionRepository;
+    @Autowired
+    AnswerRepository answerRepository;
+    @Autowired
+    QuestionController questionController;
+
+    @CrossOrigin()
+    @PostMapping("questions/{questionId}/answers")
+    protected Answer addAnswer(
+            @PathVariable Long questionId,
+            @RequestBody @Valid AnswerInput answerinput
+    ) {
```

Comment:
1. Controller shouldn't use another controller as a dependency.
2. If you create answer and set question as parent, it should create relation automatically. So there shouldn't be a need to add answer to question.answers.